rakelib/version.rake

```diff
@@ -12,15 +14,28 @@ def get_versions
 end
 
 # Update the version file, keeping the comments in tact
-def update_version_file(hash)
-  existing_versions = YAML.safe_load(File.read(VERSION_FILE))
+def update_version_file(hash, existing_versions)
   versions_as_text = IO.read(VERSION_FILE)
   %w(logstash logstash-core logstash-core-plugin-api).each do |field|
     versions_as_text.gsub!(/(?<=#{field}: )#{existing_versions[field]}/, "#{hash[field]}")
   end
   IO.write(VERSION_FILE, versions_as_text)
 end
 
+def update_index_shared1(hash, existing_versions)
```

Comment:
Nit:

Instead of passing the new_version and existing_versions hash, you could change this method to be 
`update_index_shared1(old_version, new_version)`, which may improve the readability of these helper methods a little